### DISCUSSION TITLE
Is there a way to have the information display in 2 columns?

### DISCUSSION THREAD

**Discussion by Vib1240n** [fastfetch-cli/fastfetch#1266] [Q&A]
<img width="931" alt="Screenshot 2024-09-16 at 1 57 53 PM" src="https://github.com/user-attachments/assets/1abf4802-385f-4c55-84e8-6aabf80c10d4">
Is there a way to have the Hardware info and software info be displayed right next to each other instead of top/bottom? Like 1 columns for Hardware information and right next to it 2nd column be software information.

**Comment by CarterLi** [comment]
No native support, but hack exists. You run fastfetch twice with larger padding for the second one.

![image](https://github.com/user-attachments/assets/6c6ba1d2-e143-4b97-b3dc-13ffe26d0775)

`clear;fastfetch -c left.jsonc;print -n "\e[H";fastfetch -c right.jsonc`

**Reply by RomualdoM** [reply]
It went wrong for me. When I gave the command, the following error message appeared: 
Error: couldn't find config: left.jsonc
Usage: file [-bcCdEhikLlNnprsSvzZ0] [--apple] [--extension] [--mime-encoding]
            [--mime-type] [-e <testname>] [-F <separator>]  [-f <namefile>]
            [-m <magicfiles>] [-P <parameter=value>] [--exclude-quiet]
            <file> ...
       file -C [-m <magicfiles>]
       file [--help]
Warning: unknown mime-type for "-n" -- using "application/octet-stream"
Error: no such file "-n"
Error: no such file "\e[H"
Error: couldn't find config: right.jsonc

**Comment by CarterLi** [comment]
```jsonc
// left.jsonc
{
    "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
    "display": {
        "separator": "  ",
        "constants": [
            "─────────────────" // {$1}, used in Custom module
        ],
        "key": {
            "type": "icon",
            "paddingLeft": 2
        }
    },
    "modules": [
        {
            "type": "custom", // HardwareStart
            // {#1} is equivalent to `\u001b[1m`. {#} is equivalent to `\u001b[m`
            "format": "┌{$1} {#1}Hardware Information{#} {$1}┐"
        },
        "host",
        "cpu",
        "gpu",
        "disk",
        "memory",
        "swap",
        "display",
        "brightness",
        "battery",
        "poweradapter",
        "bluetooth",
        "sound",
        "gamepad",
        "break",
        "break",
        "break",
        "break",
        "break",
        "break",
        {
            "type": "custom", // InformationEnd
            "format": "└{$1}──────────────────────{$1}┘"
        },
        {
            "type": "colors",
            "paddingLeft": 2,
            "symbol": "circle"
        }
    ]
}
```

```jsonc
// right.jsonc
{
    "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
    "logo": {
        "padding": {
            "right": 64
        }
    },
    "display": {
        "separator": "  ",
        "constants": [
            "─────────────────" // {$1}, used in Custom module
        ],
        "key": {
            "type": "icon",
            "paddingLeft": 2
        }
    },
    "modules": [
        {
            "type": "custom", // SoftwareStart
            "format": "┌{$1} {#1}Software Information{#} {$1}┐"
        },
        {
            "type": "title",
            "keyIcon": "",
            "key": "Title", // Title module has no key by default, so that icon is not displayed
            "format": "{user-name}@{host-name}"
        },
        "os",
        "kernel",
        "lm",
        "de",
        "wm",
        "shell",
        "terminal",
        "terminalfont",
        "theme",
        "icons",
        "wallpaper",
        "packages",
        "uptime",
        {
            "type": "localip",
            "compact": true
        },
        {
            "type": "wifi",
            "format": "{ssid}"
        },
        "locale",
        {
            "type": "custom", // InformationEnd
            "format": "└{$1}──────────────────────{$1}┘"
        },
        "break"
    ]
}
```

**Reply by Vib1240n** [reply]
Oh sweet, thank you very much

**Reply by Vib1240n** [reply]
I am running into this error

```
Error: couldn't find config: right.jsonc
Error: couldn't find config: left.jsonc
```
My file structure is 
```
/Users/bb8/.config/fastfetch
├── config.jsonc
├── left.jsonc
└── right.jsonc
``` 
and I have this in my .zshrc at the top - `﻿﻿﻿fastfetch -c left.jsonc; print -n "\e[H"; fastfetch -c right.jsonc` and If i run fastfetch from ~/.config/fastfetch/ then I get it running but if I run it from any other dir I get the error.

**Reply by froux07** [reply]
You have to put your jsonc file in the /usr/share/fastfetch/presets directory, or given the full path of your jsonc file.
In your case : fastfetch -c /Users/bb8/.config/fastfetch/left.jsonc ...
